### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by HildarTheDorf**
On my machine a number of tests fail. This pull request should fix the logic in the tests to remove the false negatives.

### CODE REVIEW

File: tests/unit/graphics_library_positive.cpp [hunk 1]
```diff
@@ -424,7 +424,8 @@ TEST_F(PositiveGraphicsLibrary, NotAttachmentDynamicBlendEnable) {
 
     if (!extended_dynamic_state3_features.extendedDynamicState3ColorBlendEnable ||
         !extended_dynamic_state3_features.extendedDynamicState3ColorBlendEquation ||
-        !extended_dynamic_state3_features.extendedDynamicState3ColorWriteMask) {
+        !extended_dynamic_state3_features.extendedDynamicState3ColorWriteMask ||
+        !extended_dynamic_state3_features.extendedDynamicState3ColorBlendAdvanced) {
```
Comment (spencer-lunarg) [inline tests/unit/graphics_library_positive.cpp:428]: this one is nasty, we added `VK_DYNAMIC_STATE_COLOR_BLEND_ADVANCED_EXT` to the VU a while ago and I thought I scrubbed all things effected

File: tests/unit/descriptor_buffer.cpp [hunk 2]
```diff
@@ -1348,7 +1348,9 @@ TEST_F(NegativeDescriptorBuffer, Various) {
         }
     }
 
-    {
+    auto descriptor_buffer_features = LvlInitStruct<VkPhysicalDeviceDescriptorBufferFeaturesEXT>();
```
Comment (spencer-lunarg) [inline tests/unit/descriptor_buffer.cpp:1351]: I realize in our `InitBasicDescriptorBuffer` only checks for `descriptorBuffer`, this is a good way to check, thanks

File: tests/unit/dynamic_state.cpp [hunk 3]
```diff
@@ -3769,7 +3769,7 @@ TEST_F(NegativeDynamicState, DiscardRectanglesVersion) {
     if (!AreRequiredExtensionsEnabled()) {
         GTEST_SKIP() << RequiredExtensionsNotSupported() << " not supported";
     }
-    if (InstanceExtensionSupported(VK_EXT_DISCARD_RECTANGLES_EXTENSION_NAME, 2)) {
+    if (DeviceExtensionSupported(VK_EXT_DISCARD_RECTANGLES_EXTENSION_NAME, 2)) {
```
Comment (spencer-lunarg) [inline tests/unit/dynamic_state.cpp:3772]: wow, this explains a recent failure I had

**Review by spencer-lunarg — APPROVED** [review]
LGTM! This is a really nice cleanup on these tests

**Review by juan-lunarg — APPROVED** [review]
💯

**Comment by juan-lunarg** [comment]
> On my machine a number of tests fail.

Out of curiosity what GPU are you using?

**Pull request merged**